src/coreclr/vm/mlinfo.cpp

```diff
@@ -1564,6 +1646,21 @@ MarshalInfo::MarshalInfo(Module* pModule,
         {
             TypeHandle sigTH = sig.GetTypeHandleThrowing(pModule, pTypeContext);
 
+            if (sigTH.GetMethodTable()->IsValueType())
```

Comment:
Is this change needed?

Reply:
I can revert this if we want, but it was really hard to follow the logic beforehand.

I don't even know if this code path is even needed any more ever since we got rid of WinRT support as that was the only scenario where we could get a generic down this path.

Reply:
Is this change supposed to be just a refactoring that does not change any behavior? If yes, I am fine with keeping it. (It was not obvious to me that it is not changing any behavior.)

Reply:
Yes this is a refactoring that doesn't change behavior. It was required beforehand with the way I was implementing the "runtime marshalling disabled" switch in this method, but now it's just a refactoring.